tests/embodied/motionmodel/test_misc.py

```diff
@@ -0,0 +1,61 @@
+import theseus as th
+import torch
+
+from theseus.utils import check_jacobians
+
+
+def test_nonholonomic():
```

Comment:
When we add any new custom costs like nonholonomic or hinge do we need to also test our vectorization for them or does the current unit test cover any derived classes on CostFunction?

Reply:
The planner example uses vectorization, so this should be working fine. I think adding a unit test for every such cost function might be overkill (as long as we always have some examples to test it).